predicators/main.py

```diff
@@ -212,6 +212,7 @@ def _generate_interaction_results(
     logging.info("Generating interaction results...")
     results = []
     query_cost = 0.0
+    last_state = None
```

Comment:
thanks for making this change! can we give this a try?

>if we do add that method, then we might as well just call state = env.get_state() here instead of passing the init_state in as an argument

so in particular:
1. remove the current changes in `main.py`
2. rather than `init_state`, make the new argument to `run_policy` something like `do_state_reset: bool = False`
3. then the only change in `main.py` can be `do_state_reset=(not CFG.online_learning_lifelong)`
4. change the implementation of `run_policy` accordingly
5. add a comment that the `request.train_task_idx` is ignored if `do_state_reset` is True